airflow/utils/timeout.py

```diff
@@ -22,12 +22,43 @@
 from airflow.utils.platform import IS_WINDOWS
 from airflow.exceptions import AirflowTaskTimeout
 from airflow.utils.log.logging_mixin import LoggingMixin
+from typing import ContextManager, Type
 
-TIMER_THREAD_ATTR = '_timer_thread'
+_timeout = ContextManager[None]
 
 
-class timeout(LoggingMixin):  # pylint: disable=invalid-name
-    """To be used in a ``with`` block and timeout its content."""
+class _timeout_windows(_timeout, LoggingMixin):
+
+    def __init__(self, seconds=1, error_message='Timeout'):
+        super().__init__()
+        self._timer: Timer = None
```

Comment:
I don’t think mypy would allow this

Reply:
Can you elaborate why?

Reply:
When youi have `self._timer: Timer`, the right hand side of the assignment must evaluate to a `Timer`, which `None` is not. To make Mypy accept this, you need to annotate the type as `Optional[Timer]` instead.

Reply:
If you [set up pre-commit](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#contribution-workflow), this (and other linter errors in this PR) can be more easily caught before you push the changes.

Reply:
Thanks for explaining. I was not familiar with mypy but this seems very useful also for future projects :)
I have commited the change request and there is now an Optional annotation.